include/grpc/grpc_security.h

```diff
@@ -301,6 +301,20 @@ GRPCAPI grpc_call_credentials* grpc_composite_call_credentials_create(
 GRPCAPI grpc_call_credentials* grpc_google_compute_engine_credentials_create(
     void* reserved);
 
+/** Creates compute engine channel credentials to connect to a google gRPC
+   service.
+
+   This channel credential is expected to be used within a composite credential
+   alongside a compute_engine_credential. If used in conjunction with any call
+   credential besides a compute_engine_credential, the connection may suddenly
+   and unexpectedly begin to fail RPCs.
+
+   WARNING: Do NOT use this credentials to connect to a non-google service as
+   this could result in an oauth2 token leak. The security level of the
+   resulting connection is GRPC_PRIVACY_AND_INTEGRITY. */
+GRPCAPI grpc_channel_credentials*
+grpc_compute_engine_channel_credentials_create(void* reserved);
```

Comment:
This is an addition to the C-core API.  It needs either a gRFC or to be marked as experimental.

Reply:
Understood. I actually marked the Python-level API as experimental. Makes sense to do that here as well.

Reply:
Experimental status added to docstring.